### PR TITLE
Add 'id token from external servers' integration test

### PR DESCRIPTION
**Pull request by dmitrizagidulin**

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
Adding a minimal integration test using an externally-issued id token on a different pod, using an http rest client. /cc @timbl @dan-f

File: test/integration/acl-oidc.js [hunk 1]
```diff
@@ -305,6 +307,44 @@ describe('ACL HTTP', function () {
...
+    it('client should use an externally issued id token', () => {
+      let decodedToken = JWT.decode(idToken)
+      let tokenIssuer = decodedToken.payload.iss
+      let tokenWebId = decodedToken.payload.sub
+
+      assert.equal(tokenIssuer, 'https://example.com')
+      assert.equal(tokenWebId, 'https://alice.example.com/profile/card#me')
+    })
+
+    it('should recognize an external server id token', (done) => {
```
Comment (dan-f) [inline test/integration/acl-oidc.js]: This test isn't what I took away from our discussion. I'm seeing a unit test where the identity provider is faked out. I'd expect, e.g. `example.com`, to be an actual solid server just like the one behind `timAccountUri`. At least, that's what integration means to me in this context.

File: test/integration/acl-oidc.js [hunk 2]
```diff
@@ -305,6 +307,44 @@ describe('ACL HTTP', function () {
...
+
+    const externalServer = path.join(__dirname,
+      '../resources/external-servers/example.com')
+    let testConfig = require(path.join(externalServer, 'openid-configuration.json'))
+    let testJwks = require(path.join(externalServer, 'jwks.json'))
+
+    nock('https://example.com').get('/jwks').reply(200, testJwks)
+    nock('https://example.com').get('/.well-known/openid-configuration').reply(200, testConfig)
+
+    it('client should use an externally issued id token', () => {
```
Comment (dan-f) [inline test/integration/acl-oidc.js]: Looks like this just verifies that the [anvilresearch/jose](https://github.com/anvilresearch/jose/) library knows how to parse a JWT. I don't see why this needs to be added.

Comment (dmitrizagidulin) [inline test/integration/acl-oidc.js]: No prob, will remove.

**Comment by dmitrizagidulin** [comment]
> I was hoping to see tests against a system of a client, identity provider, and resource provider. Integration tests at that level could verify what the client receives depending on the various things that could succeed or go wrong during a handshake.

Agreed. This is a first step though, just to demonstrate that the ACL system will accept tokens from external servers presented by a REST client. I'll also be adding more elaborate integration tests like you describe (with 3 express servers interacting). Though if at all possible, I'd like to add those after we do our integration demo?

**Comment by dan-f** [comment]
There's definitely value to a reasonable ratio of `unit : integration` tests. This one looks much more like a unit test. Maybe then put it with the rest of the unit tests?

**Comment by dmitrizagidulin** [comment]
I see what you're saying. I put it here with the integration test, tho, because it still requires a running solid server, in addition to the one stubbed by nock.

**Comment by dmitrizagidulin** [comment]
Closing. Superceded by PR #509.

**Pull request closed without merging**